### PR TITLE
Orca Type support and utility operation on Spark

### PR DESCRIPTION
**Pull request by jenniew**
Provide utility operation to run customized function, return value or exception.
Create type structure and save class name  when creating SparkXShards.
type = {'class_name': class_name,
             'schema': schema}
Provide _get_schema operation to get schema and save to type structure. 
Related issue: https://github.com/analytics-zoo/orca/issues/25
https://github.com/analytics-zoo/orca/issues/42

### CODE REVIEW

File: pyzoo/zoo/orca/data/shard.py [hunk 1]
```diff
@@ -261,3 +292,25 @@ def load_pickle(cls, path, sc, minPartitions=None):
 
     def __del__(self):
         self.uncache()
+
+    def _utility(self, func, *args, **kwargs):
```
Comment (jason-dai) [inline pyzoo/zoo/orca/data/shard.py:296]: maybe change to `_for_each`?

File: pyzoo/zoo/orca/data/shard.py [hunk 2]
```diff
@@ -110,11 +110,41 @@ def get_data(self):
...
+        return self.data
+
+
 class SparkXShards(XShards):
     def __init__(self, rdd):
         self.rdd = rdd
         self.user_cached = False
         self.rdd.cache()
+        self.type = {}
+        self.type['class_name'] = self._utility(get_class_name).first()
```
Comment (jason-dai) [inline pyzoo/zoo/orca/data/shard.py:147]: change to `_get_class_name`?

File: pyzoo/zoo/orca/data/shard.py [hunk 3]
```diff
@@ -150,14 +180,12 @@ def repartition(self, num_partitions):
         return repartitioned_shard
 
     def partition_by(self, cols, num_partitions=None):
-        import pandas as pd
-        elem_class, columns = self.rdd.map(
-            lambda data: (type(data), data.columns) if isinstance(data, pd.DataFrame)
-            else (type(data), None)).first()
-        if issubclass(elem_class, pd.DataFrame):
+        if self.type['class_name'] == 'pandas.core.frame.DataFrame':
```
Comment (jason-dai) [inline pyzoo/zoo/orca/data/shard.py:183]: change to `_get_class_name`?

File: pyzoo/zoo/orca/data/shard.py (hunk 2 above)
Comment (jenniew) [reply pyzoo/zoo/orca/data/shard.py:147]: ok

Comment (jenniew) [reply pyzoo/zoo/orca/data/shard.py]: updated

File: pyzoo/zoo/orca/data/shard.py (hunk 3 above)
Comment (jenniew) [reply pyzoo/zoo/orca/data/shard.py]: updated

Comment (jenniew) [reply pyzoo/zoo/orca/data/shard.py]: updated

**Comment by jason-dai** [comment]
LGTM

**Comment by jenniew** [comment]
Jenkins Pass: http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Validation/3615/

**Pull request merged**